src/Microsoft.ApplicationInsights/DataContracts/RequestTelemetry.cs

```diff
@@ -244,6 +244,37 @@ internal string MetricExtractorInfo
             set;
         }
 
+        internal RequestData Data
+        {
+            get
+            {
+                return LazyInitializer.EnsureInitialized(ref this.dataPrivate,
```

Comment:
After initialized it might be requested again and will need to be updated. Unless you absolutely sure nobody will change any property between two calls to `Data`

Reply:
As a minimum leave a comment about this